Compare-NET-Objects/TypeComparers/ListComparer.cs

```diff
@@ -146,17 +151,48 @@ private bool ListsHaveDifferentCounts(CompareParms parms)
             return false;
         }
 
-        private bool ChildIsListOrDictionary(CompareParms parms)
+        private bool ChildShouldBeComparedWithoutOrder(CompareParms parms)

```

Comment:
This check does not work for my case because nested collection starts with null. Now we take into account all values in collection.